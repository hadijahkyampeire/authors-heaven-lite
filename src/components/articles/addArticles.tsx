import React, { useState } from 'react';
import { ApiArticle } from 'types/articles';
import { createArticle } from 'actions/articles';
import { AddArticleForm } from 'components/forms';
import { Redirect } from 'react-router';
import { ArticleData } from 'types/articles';
import { Header } from "components/commons";

import './__styles__/addArticle.scss';

interface Props {
  onSubmit: (
    ...args: Parameters<typeof createArticle>
  ) => void;
  newArticle?: ApiArticle;
}
export const AddArticle: React.FC<Props> = ({ newArticle, onSubmit }) => {
  const initialValues: ArticleData = {
    title: "",
    description: "",
    body: "",
    favourited: false,
    published: false
    
  };

  const [form, setForm] = useState<ArticleData>(initialValues);

  const publishedForm = {...form, published: true };
  const handlePublish = (e: any) => { e.preventDefault(); onSubmit(publishedForm); };
  const handleSave = (e: any) => { e.preventDefault(); onSubmit(form); };
 
  if (newArticle) {
    return <Redirect to={newArticle.published === true ? '/articles/published/' : '/articles/'} />
  }
  return (
    <section className="add-article-page">
      <Header title='Create Article'/>
      <AddArticleForm newArticle={newArticle} onPublish={handlePublish} onSave={handleSave} form={form} setForm={setForm} />
    </section>
  );
};
