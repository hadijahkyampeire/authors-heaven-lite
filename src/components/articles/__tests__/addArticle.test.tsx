import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render } from "testUtils";

import { AddArticle } from '../addArticles';

type addArticleProps = React.ComponentProps<typeof AddArticle>;

const baseProps: addArticleProps = {
  onSubmit: jest.fn()
};

function renderUI(props: Partial<addArticleProps> = {}) {
  return render(
    <Router>
      <AddArticle {...baseProps} {...props} />
    </Router>
  );
};

describe('<AddArticle />', () => {
  it("snapshot test", () => {
    const { container } = renderUI({});
    expect(container).toMatchSnapshot();
  });

  it('renders properly', async () => {
    const { getByText, findByTestId } = renderUI({});

    const articleForm = await findByTestId('article-form');
    expect(getByText('Create Article')).toBeInTheDocument();
    expect(articleForm).toBeInTheDocument();
  });

  it("should have a form with input field", async () => {
    const { findByTestId } = renderUI({});

    const articleForm = await findByTestId('article-form');
    expect(articleForm).toHaveFormValues({
      title: "",
      description: "",
      body: ""
    });
  });
});
