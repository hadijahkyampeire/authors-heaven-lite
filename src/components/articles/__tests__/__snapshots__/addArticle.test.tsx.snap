// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddArticle /> snapshot test 1`] = `
<div>
  <section
    class="add-article-page"
  >
    <div
      class="app-header"
    >
      <span
        class="bx--tooltip--definition bx--tooltip--a11y"
      >
        <button
          aria-describedby="definition-tooltip-1"
          class="bx--tooltip__trigger bx--tooltip--a11y bx--tooltip__trigger--definition bx--tooltip--top bx--tooltip--align-start"
          type="button"
        >
          <div>
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="32"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 26L15.41 24.59 7.83 17 28 17 28 15 7.83 15 15.41 7.41 14 6 4 16 14 26z"
              />
            </svg>
          </div>
        </button>
        <span
          class="bx--assistive-text"
          id="definition-tooltip-1"
          role="tooltip"
        >
          Go Back
        </span>
      </span>
      <div
        class="title"
      >
        Create Article
      </div>
    </div>
    <form
      class="bx--form"
      data-testid="article-form"
    >
       
      <div
        style="margin-bottom: 2rem;"
      >
        <div
          class="bx--form-item bx--text-input-wrapper"
        >
          <label
            class="bx--label"
            for="title"
          >
            Title
          </label>
          <div
            class="bx--text-input__field-outer-wrapper"
          >
            <div
              class="bx--text-input__field-wrapper"
            >
              <input
                class="bx--text-input bx--text__input"
                id="title"
                name="title"
                placeholder="Enter article title"
                required=""
                title="Enter article title"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        style="margin-bottom: 2rem;"
      >
        <div
          class="bx--form-item bx--text-input-wrapper"
        >
          <label
            class="bx--label"
            for="description"
          >
            Description
          </label>
          <div
            class="bx--text-input__field-outer-wrapper"
          >
            <div
              class="bx--text-input__field-wrapper"
            >
              <input
                class="bx--text-input bx--text__input"
                id="description"
                name="description"
                placeholder="Enter description"
                required=""
                title="Enter description"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        style="margin-bottom: 2rem;"
      >
        <div
          class="bx--form-item"
        >
          <label
            class="bx--label"
            for="body"
          >
            Body
          </label>
          <div
            class="bx--text-area__wrapper"
          >
            <textarea
              class="bx--text-area"
              cols="50"
              id="body"
              name="body"
              placeholder="Enter article body"
              rows="4"
            />
          </div>
        </div>
      </div>
      <div
        class="action-buttons"
      >
        <button
          class="button cancel-create-btn"
        >
          Cancel
        </button>
        <button
          class="bx--btn bx--btn--secondary submit-article bx--btn bx--btn--primary"
          id="create-as-draft"
          tabindex="0"
          type="button"
        >
          Save as Draft
        </button>
        <button
          class="bx--btn bx--btn--primary submit-article bx--btn bx--btn--primary"
          id="create-and-publish"
          tabindex="0"
          type="button"
        >
          Submit And Publish
        </button>
      </div>
       
    </form>
  </section>
</div>
`;
